src/debug.cc

```diff
@@ -210,38 +272,29 @@ static void
 debugOpenLog(const char *logfile)
 {
     if (logfile == NULL) {
-        debug_log = stderr;
+        TheLog.clear();
         return;
     }
 
-    if (debug_log_file)
-        xfree(debug_log_file);
-
-    debug_log_file = xstrdup(logfile);  /* keep a static copy */
-
-    if (debug_log && debug_log != stderr)
-        fclose(debug_log);
-
     // Bug 4423: ignore the stdio: logging module name if present
     const char *logfilename;
     if (strncmp(logfile, "stdio:",6) == 0)
         logfilename = logfile + 6;
     else
         logfilename = logfile;
 
-    debug_log = fopen(logfilename, "a+");
-
-    if (!debug_log) {
+    if (auto log = fopen(logfilename, "a+")) {
+#if _SQUID_WINDOWS_
+        setmode(fileno(log), O_TEXT);
+#endif
+        TheLog.reset(log, logfilename);
+    } else {
         fprintf(stderr, "WARNING: Cannot write log file: %s\n", logfile);
         perror(logfile);
         fprintf(stderr, "         messages will be sent to 'stderr'.\n");
         fflush(stderr);
-        debug_log = stderr;
+        TheLog.clear();
```

Comment:
This is the line I was referring to. IMO this else case should be redone to to following sequence:
{
 // record the problem in the previous log output location
 debugs(1, DBG_CRITICAL, "ERROR: Cannot write log file: " << logfile);
 TheLog.clear(); // or StopUsingDebugLog();
 // also record the problem to the emergency log output now being used
 debugs(1, DBG_CRITICAL, "ERROR: Cannot write log file: " << logfile);
}